packages/loot-core/src/server/schedules/app.ts

```diff
@@ -274,13 +282,13 @@ export async function updateSchedule({
   if (schedule.rule) {
     throw new Error('You cannot change the rule of a schedule');
   }
-
-  if (schedule.name) {
-    if (await checkIfScheduleExists(schedule.name, schedule.id)) {
+  let scheduleName = await nameFromScheduleId(schedule.id);
+  if (scheduleName) {
+    if (await checkIfScheduleExists(scheduleName, schedule.id)) {
       throw new Error('There is already a schedule with this name');
     }
   } else {
-    schedule.name = null;
+    scheduleName = null;
   }
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏why do we even need this `else` condition? If we remove it: would that fix the problem?

Reply:
You're right, I thought that was so that we could catch duplicate schedule names but we check that elsewhere anyways.